lib/embedded-linux-pipeline.ts

```diff
@@ -236,11 +262,22 @@ export class EmbeddedLinuxPipelineStack extends cdk.Stack {
       project,
     });
 
-    const artifactAction = new codepipeline_actions.S3DeployAction({
-      actionName: 'Artifact',
-      input: buildOutput,
-      bucket: outputBucket,
-    });
+    let artifactAction: codepipeline_actions.S3DeployAction;
+
+    if (props.subDirectoryName){
+      artifactAction = new codepipeline_actions.S3DeployAction({
```

Comment:
Only some things get into the final bucket with this action. There is also some things in the build spec files.

Reply:
What do you mean by "some things" ?
By setting the subDirectoryName all uploaded artifacts go into a sub dir...